File: Baekjoon/solved/3687.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

let input = [];

readline.on('line', function(line) {
    input.push(line)
}).on('close', function(){
    const t = Number(input[0])
    // d = [남은 성냥갯수] = 숫자.
    let dp = Array.from(new Array(101), ()=> Infinity)
	dp[2] = 1
    dp[3] = 7
    dp[4] = 4
    dp[5] = 2
    dp[6] = 6
    dp[7] = 8
    dp[8] = 10

    for(let num=9; num<101; num++) {
        
    }


    for(let i=0; i<t; i++) {
        const n = Number(input[i+1])
       
    }

    process.exit()
})

File: Baekjoon/solved/3687.py
import sys
 
def input():
    return sys.stdin.readline()
 
N = int(input())
INF = float('inf')
dp = [INF for _ in range(101)]
dp[2] = 1
dp[3] = 7
dp[4] = 4
dp[5] = 2
dp[6] = 6
dp[7] = 8
dp[8] = 10
 
for num in range(9,101):
    for i in range(2,8):
        if i != 6:
            dp[num] = min(dp[num],dp[num-i]*10+dp[i])
        else:
            dp[num] = min(dp[num],dp[num-i]*10)
for _ in range(N):
    k = int(input())
    
    max_value = ('7' if k%2 else '1' ) +'1'*(k//2-1)
    print(dp[k],max_value)